packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -0,0 +1,441 @@
+        if (installed) {
+            return <button className="theia-button action" onClick={this.uninstall}>Uninstall</button>;
+        }
+        return <button className="theia-button prominent action" onClick={this.install}>Install</button>;
+    }
+
+}
+export namespace AbstractVSXExtensionComponent {
+    export interface Props {
+        extension: VSXExtension
+    }
+}
+
+const downloadFormatter = new Intl.NumberFormat();
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const downloadCompactFormatter = new Intl.NumberFormat(undefined, { notation: 'compact', compactDisplay: 'short' } as any);
+
+export class VSXExtensionComponent extends AbstractVSXExtensionComponent {
+    render(): React.ReactNode {
+        const { iconUrl, publisher, displayName, description, version, downloadCount, averageRating } = this.props.extension;
+        const iconStyle: React.CSSProperties = {};
+        if (iconUrl) {
+            iconStyle.backgroundImage = `url('${iconUrl}')`;
+        }
+        return <div className='theia-vsx-extension'>
+            <div className='theia-vsx-extension-icon' style={iconStyle} />
+            <div className='theia-vsx-extension-content'>
+                <div className='title'>
+                    <div className='noWrapInfo'>
+                        <span className='name'>{displayName}</span> <span className='version'>{version}</span>
+                    </div>
+                    <div className='stat'>
+                        {downloadCount && <span className='download-count'><i className='fa fa-download' />{downloadCompactFormatter.format(downloadCount)}</span>}
+                        {averageRating && <span className='average-rating'><i className='fa fa-star' />{averageRating.toFixed(1)}</span>}
+                    </div>
+                </div>
+                <div className='noWrapInfo theia-vsx-extension-description'>{description}</div>
+                <div className='theia-vsx-extension-action-bar'>
+                    <span className='noWrapInfo theia-vsx-extension-publisher'>{publisher}</span>
+                    {this.renderAction()}
+                </div>
+            </div>
+        </div >;
+    }
+}
+
+export class VSXExtensionEditorComponent extends AbstractVSXExtensionComponent {
+    render(): React.ReactNode {
+        const { builtin, preview, id, iconUrl, publisher, displayName, description, averageRating, downloadCount, repository, license, readme } = this.props.extension;
+        const iconStyle: React.CSSProperties = {};
+        if (iconUrl) {
+            iconStyle.backgroundImage = `url('${iconUrl}')`;
+        }
+        return <React.Fragment>
+            <div className='header'>
+                <div className='icon-container' style={iconStyle} />
+                <div className='details'>
+                    <div className='title'>
+                        <span title='Extension name' className='name' onClick={this.openExtension}>{displayName}</span>
```

Comment:
A not-so-nice side effect of using `onClick` handlers instead of `<a>` elements is that the user cannot see where the link is going. I find it a bit confusing in this case because there are so many `onClick` links and they all lead to different targets.

Reply:
There are different css classes are applied by default to elements in theia to how render them active, focused or visited. It would require undoing all of them.